actions/mongodb_explain_action.go

```diff
@@ -0,0 +1,81 @@
+// pmm-agent
+// Copyright (C) 2018 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package actions
+
+import (
+	"context"
+	"strings"
+
+	"github.com/percona/percona-toolkit/src/go/mongolib/explain"
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type mongodbExplainAction struct {
+	id     string
+	params *agentpb.StartActionRequest_MongoDBExplainParams
+}
+
+// NewPostgreSQLShowCreateTableAction creates PostgreSQL SHOW CREATE TABLE Action.
+// This is an Action that can run `\d+ table` command analog on PostgreSQL service with given DSN.
+func NewNomgoDBExplain(id string, params *agentpb.StartActionRequest_MongoDBExplainParams) Action {
+	return &mongodbExplainAction{
+		id:     id,
+		params: params,
+	}
+}
+
+// ID returns an Action ID.
+func (a *mongodbExplainAction) ID() string {
+	return a.id
+}
+
+// Type returns an Action type.
+func (a *mongodbExplainAction) Type() string {
+	return "postgresql-show-create-table"
+}
+
+// Run runs an Action and returns output and error.
+func (a *mongodbExplainAction) Run(ctx context.Context) ([]byte, error) {
+	dsn := a.params.Dsn
+	if !strings.HasPrefix(dsn, "mongodb://") {
```

Comment:
As a general rule, pmm-agent is "dumb", pmm-managed is "smart". We can trust pmm-managed to send valid DSN.